02-express-tutorial/controlers/people.js

```diff
@@ -0,0 +1,59 @@
+let { people } = require("../data");
+
+const getPeople = (req, res) => {   
+    res.status(200).json({ success: true, data: people });
+}
+const createPerson = (req, res) => {
+    const { name } = req.body;
+    if(!name){
+        return res
+            .status(400)
+            .json({ success: false, msg: "Please provide name value" });
+    }
+    res.status(201).json({ success: true, person: name }); 
+}
+const createPersonPostman =  (req, res) => {
+    const { name } = req.body;
+    if(!name){
+        return res
+            .status(400)
+            .json({ success: false, msg: "Please provide name value" });
+    }
+    res.status(201).json({ success: true, data: [...people, name] }); 
```

Comment:
Requires change:
This can be persisted via mutating the `people` array (this only mutates the memory, not the actual file)
```js
people.push({ id: people.length + 1, name: req.body.name });

res.status(201).json({ success: true, data: people });
```